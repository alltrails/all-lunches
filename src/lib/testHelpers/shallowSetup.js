import React from 'react';
import { shallow } from 'enzyme';

export default (Component, defaultProps) =>
  (otherProps = {}) => {
    const props = { ...defaultProps, ...otherProps };
    const wrapper = shallow(<Component {...props} />);

    return { props, wrapper };
  };
