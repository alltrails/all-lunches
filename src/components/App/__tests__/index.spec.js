/* eslint-env jest */

import shallowSetup from 'lib/testHelpers/shallowSetup';

import { App } from '../index';

describe('App', () => {
  const defaultProps = { error: null, isLoading: false };
  const setup = shallowSetup(App, defaultProps);

  it('renders the expected view', () => {
    const { wrapper } = setup();
    expect(wrapper).toMatchSnapshot();
  });

  it('renders the expected view when loading', () => {
    const { wrapper } = setup({ isLoading: true });
    expect(wrapper).toMatchSnapshot();
  });

  it('renders the expected view when there is an error', () => {
    const { wrapper } = setup({ error: 'looks like there was an error' });
    expect(wrapper).toMatchSnapshot();
  });
});
