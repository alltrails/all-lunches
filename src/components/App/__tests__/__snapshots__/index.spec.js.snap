// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders the expected view 1`] = `<Dashboard />`;

exports[`App renders the expected view when loading 1`] = `
<style__LoadingWrapper>
  <style__LoadingTitle>
    Loading
  </style__LoadingTitle>
  <LoadingSpinner
    color="#428813"
  />
</style__LoadingWrapper>
`;

exports[`App renders the expected view when there is an error 1`] = `
<ErrorView
  hasBackground={true}
  text="looks like there was an error"
/>
`;
